src/blockdata/witness.rs

```diff
@@ -211,6 +213,17 @@ impl Witness {
         self.content[end_varint..].copy_from_slice(new_element);
     }
 
+    /// Pushes a DER-encoded ECDSA signature with a signature hash type as a new element on the
+    /// witness, requires an allocation.
+    pub fn push_bitcoin_signature(&mut self, signature: &ecdsa::SerializedSignature, hash_type: EcdsaSighashType) {
```

Comment:
don't we want this for Schnorr too?

or maybe a generic one:

```rust
pub fn push_with_hash_type<T: AsRef<[u8]>>(&mut self, new_element: T, hash_type: u8) {
```

Reply:
+1 to having a totally separate method for schnorr (doesn't need to be in this PR)